sway/desktop/layer_shell.c

```diff
@@ -115,25 +115,35 @@ static void arrange_layer(struct sway_output *output, struct wl_list *list,
 		// Horizontal axis
 		const uint32_t both_horiz = ZWLR_LAYER_SURFACE_V1_ANCHOR_LEFT
 			| ZWLR_LAYER_SURFACE_V1_ANCHOR_RIGHT;
+		const bool match_both_horiz =
+			(state->anchor & ZWLR_LAYER_SURFACE_V1_ANCHOR_LEFT) &&
+			(state->anchor & ZWLR_LAYER_SURFACE_V1_ANCHOR_RIGHT);
 		if ((state->anchor & both_horiz) && box.width == 0) {
 			box.x = bounds.x;
 			box.width = bounds.width;
-		} else if ((state->anchor & ZWLR_LAYER_SURFACE_V1_ANCHOR_LEFT)) {
+		} else if ((state->anchor & ZWLR_LAYER_SURFACE_V1_ANCHOR_LEFT)
+				&& !match_both_horiz) {
```

Comment:
How about just adding another both left + right case above for box.width != 0? I think it makes sense to preserve the simple "both -> left -> right -> neither" flow.

I don't like the use of "both_horiz" in this function, since it sort of seems like state->anchor & both_horiz should only be truthful if the anchor contains both horizontal borders, but it's actually true if the anchor contains either. So I think you could just replace the definition of `both_horiz` with your `match_both_horiz` and the whole block would make more sense.

Reply:
Sure, but that would mean duplicating the averaging code.

Reply:
I don't think that's a problem – it's only one line. It makes more sense to me to keep all the cases clear.